main/lsp/requests/code_action.cc

```diff
@@ -260,6 +262,27 @@ unique_ptr<ResponseMessage> CodeActionTask::runRequest(LSPTypecheckerDelegate &t
             action->kind = CodeActionKind::RefactorRewrite;
             action->edit = move(workspaceEdit);
             result.emplace_back(move(action));
+        } else {
+            if (loc.beginPos() != loc.endPos()) {
+                // For move method to new module we use canResolveLazily to defer the computation
+                // until the user has actually selected the action. We can't do that here because
+                // we need to do the core computation to know if extract the current selection is
+                // valid in the first place, to decide if we can show the code action or not.
```

Comment:
Let's add a Timer here. I'm still worried that this will be too slow, and the Timer will tell us whether it is or isn't